### PR TITLE
Changes to StrideInfo and some fixes to tiling and resize

### PR DESCRIPTION
**Pull request by jbruestle**
Updates to StrideInfo, Tiling and ResizeTmps so that they work on non-canonical affine loops.

### CODE REVIEW

**Comment by jbruestle** [comment]
@shiyao00 Wanted to give you a heads up since I changed the semantics of tile for cases with non-1 step sizes (provided tile size is the total range not the relative range).  I'm not sure if any internal code relies on those semantics, but I noticed you added a test for that case.

File: pmlc/dialect/pxa/analysis/strides.cc [hunk 1]
```diff
@@ -28,27 +29,44 @@ static std::string getUniqueName(Block *ref, BlockArgument arg) {
       .str();
 }
 
+// Generally useful helper function
+int64_t GetIVStep(BlockArgument arg) {
```
Comment (flaub) [inline pmlc/dialect/pxa/analysis/strides.cc:33]: Rename `getIVStep`

File: pmlc/dialect/pxa/analysis/strides.cc [hunk 2]
```diff
@@ -181,6 +215,14 @@ void StrideInfo::print(raw_ostream &os, Block *relative) const {
   os << ']';
 }
 
+AffineValueMap StridesToValueMap(MLIRContext *ctx, ArrayRef<StrideInfo> dims) {
```
Comment (flaub) [inline pmlc/dialect/pxa/analysis/strides.cc:218]: Not sure if we should say `stridesToValueMap` or something else.

Comment (flaub) [reply pmlc/dialect/pxa/analysis/strides.cc:218]: Like, `getStridesToValueMap` or `convertStridesToValueMap`

**Pull request merged**